spec/rails_helper.rb

```diff
@@ -149,7 +149,12 @@ def self.test_setup(x = nil)
   end
 end
 
-TestProf::BeforeAll.configure { |config| config.before(:begin) { TestSetup.test_setup } }
+TestProf::BeforeAll.configure do |config|
+  config.after(:begin) do
+    TestSetup.test_setup
+    DB.test_transaction = ActiveRecord::Base.connection.current_transaction
```

Comment:
There is another block similar to this one on line ~387. I am wondering, is doing that still necessary if you do it here while configuring the tests? 🤔

Reply:
Yes, because the group transaction and the test transaction are different. I verified this worked with a simple test that used `DB.after_commit` in a `before_all` block.